### PR TITLE
MsAuth10AtPop part2

### PR DESCRIPTION
**Pull request by jennyf19**

### CODE REVIEW

File: src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs [hunk 1]
```diff
@@ -370,15 +370,26 @@ private void LogAuthResult(AuthenticationResult? authenticationResult)
...
-                    builder.WithProofOfPosessionKeyId(tokenAcquisitionOptions.PopPublicKey, "pop");
-                    builder.OnBeforeTokenRequest((data) =>
+                    if (string.IsNullOrEmpty(tokenAcquisitionOptions.JwkClaim))
                     {
-                        data.BodyParameters.Add("req_cnf", tokenAcquisitionOptions.PopPublicKey);
-                        data.BodyParameters.Add("token_type", "pop");
-                        return Task.CompletedTask;
-                    });
+                        builder.WithProofOfPosessionKeyId(tokenAcquisitionOptions.PopPublicKey, "pop");
+                        builder.OnBeforeTokenRequest((data) =>
```
[the remark this answers is not in the document]
Comment (bgavrilMS) [reply src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs:378]: Also happy to hear how we can re-design the `PoPAuthenticationConfiguration` interface. The original design did not take into account Wilson creating the SHR. It works, but it's clunky...

File: src/Microsoft.Identity.Web.TokenAcquisition/MsAuth10AtPop.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            builder.WithProofOfPosessionKeyId(popPublicKey);
+            builder.OnBeforeTokenRequest((data) =>
+             {
+                 string? signedAssertion = GetSignedClientAssertion(
+                     clientCertificate,
+                     data.RequestUri.AbsoluteUri,
+                     jwkClaim,
+                     clientId);
+
+                 data.BodyParameters.Remove("client_assertion");
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.TokenAcquisition/MsAuth10AtPop.cs:33]: @bgavrilMS 
Would it be possible, when we use builder.WithProofOfPosessionKeyId(popPublicKey) that MSAL does not compute the signed assertion? (as it's not necessary)
I can create a GitHub issue in the MSAL.NET repo if you agree.

File: src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jennyf19) [reply src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs:378]: @jmprieur said you synced off-line and are okay with this proposal, is that right?

File: src/Microsoft.Identity.Web.TokenAcquisition/MsAuth10AtPop.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            return builder;
+        }
+
+        private static string? GetSignedClientAssertion(
+            X509Certificate2 certificate,
+            string audience,
+            string jwkClaim,
+            string clientId)
+        {
+            // no need to add exp, nbf as JsonWebTokenHandler will add them by default.
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web.TokenAcquisition/MsAuth10AtPop.cs:47]:
```suggestion
            // no need to add exp, nbf as JsonWebTokenHandler will add them by default
```

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @jennyf19 

successfully tested

**Pull request merged**